tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+      );
+    $this->assertTrue(
+    method_exists($collection, 'getAuthorizationUrl'),
+        'OAuth2ManagerInterface does not have getAuthorizationUrl function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getState'),
+        'OAuth2ManagerInterface does not have getState function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getUserInfo'),
+        'OAuth2ManagerInterface does not have getUserInfo function/method'
+      );
+
+  }
+
+  /**
+   * test for class SocialApiDataHandler
+   */
+
+  public function testSocialApiDataHandler () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/SocialApiDataHandler.php');
+    $this->session = $this->getMock(SessionInterface::class);
+    $collection = $this->getMockBuilder('Drupal\social_api\SocialApiDataHandler')
+                       ->setConstructorArgs(array($this->session))
+                       ->getMockForAbstractClass();
+
+    // checking for correct setSessionPrefix and getSessionPrefix method
+    $collection->setSessionPrefix('1234');
+    // var_dump($collection->getSessionPrefix());
+    $this->assertEquals('1234_', $collection->getSessionPrefix());
+  }
+
+  /**
+   * test for class SocialApiController
+   */
+
+  public function testSocialApiController () {
+
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/Controller/SocialApiController.php');
+
+    $this->namespaces = $this->createMock(Traversable::class);
+    $this->cache_backend = $this->createMock(CacheBackendInterface::class);
+    $this->module_handler = $this->createMock(ModuleHandlerInterface::class);
+
+    // passing parameters for constructor method in NetWorkManager Class
+   $this->networkManager = $this->getMockBuilder('Drupal\social_api\Plugin\NetworkManager')
+                                 ->setConstructorArgs(array($this->namespaces, $this->cache_backend, $this->module_handler))
+                                 ->getMock();
+
+
+   $collection = $this->getMockBuilder('Drupal\social_api\Controller\SocialApiController')
+                                ->setConstructorArgs(array($this->networkManager))
+                                ->getMock();
+
+   $this->assertTrue(
+      method_exists($collection, 'create'),
+      'SocialApiController does not have getPluginId function/method'
+    );
+
+   $this->assertTrue(
+      method_exists($collection, 'integrations'),
+      'SocialApiController does not have getPluginId function/method'
+    );
+    $expected = $collection->integrations($this->type);
+  }
+
+  /**
+   * Test for class SocialApiException
+   */
+
+  public function testSocialApiException () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/SocialApiException.php');
+    $collection = new SocialApiException();
```

Comment:
Try to do something more interesting with this, like throwing the exception and catching it, then checking for the type, the message, etc.